x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticator.java

```diff
@@ -395,4 +402,18 @@ private void checkLifetimeRestrictions(Conditions conditions) {
     private void checkLifetimeRestrictions(SubjectConfirmationData subjectConfirmationData) {
         validateNotOnOrAfter(subjectConfirmationData.getNotOnOrAfter());
     }
+
+    private void warnOnReservedAttributeNames(Assertion assertion, List<Attribute> attributes) {
+        attributes.stream().map(Attribute::getName).forEach(attributeName -> {
```

Comment:
Sanity check: we only care about these terms occurring in `name`, not in `friendlyName` or should we check both?

Reply:
Since we allow mapping using either name or friendlyName, we should check both of them.